### PR TITLE
[PF-1074] format config option

### PR DESCRIPTION
**Pull request by jaycarlton**
Persist an option called `format` in the config file. This allows a user to lock in the JSON format and avoid using `--format` for every command.

```
jaycarlton~/repos/terra-cli (jaycarlton-PF-1074) $ terra config list
[app-launch] app launch mode = DOCKER_CONTAINER
[browser] browser launch for login = AUTO
[image] docker image id = gcr.io/terra-cli-dev/terra-cli/0.116.0:stable
[resource-limit] max number of resources to allow per workspace = 1000

[logging, console] logging level for printing directly to the terminal = OFF
[logging, file] logging level for writing to files in /Users/jaycarlton/.terra/logs = INFO

[server] server = broad-dev-cli-testing
[workspace] workspace = (unset)
[text, json] output format = text
jaycarlton~/repos/terra-cli (jaycarlton-PF-1074) $ terra config set format json
Output format is json (CHANGED).
jaycarlton~/repos/terra-cli (jaycarlton-PF-1074) $ terra config list
{
  "browserLaunchOption" : "AUTO",
  "commandRunnerOption" : "DOCKER_CONTAINER",
  "dockerImageId" : "gcr.io/terra-cli-dev/terra-cli/0.116.0:stable",
  "resourcesCacheSize" : 1000,
  "fileLoggingLevel" : "INFO",
  "consoleLoggingLevel" : "OFF",
  "serverName" : "broad-dev-cli-testing",
  "workspaceId" : null,
  "format" : "json"
}
jaycarlton~/repos/terra-cli (jaycarlton-PF-1074) $ terra config set format text
Output format is text (CHANGED).
jaycarlton~/repos/terra-cli (jaycarlton-PF-1074) $ terra config list
[app-launch] app launch mode = DOCKER_CONTAINER
[browser] browser launch for login = AUTO
[image] docker image id = gcr.io/terra-cli-dev/terra-cli/0.116.0:stable
[resource-limit] max number of resources to allow per workspace = 1000

[logging, console] logging level for printing directly to the terminal = OFF
[logging, file] logging level for writing to files in /Users/jaycarlton/.terra/logs = INFO

[server] server = broad-dev-cli-testing
[workspace] workspace = (unset)
[text, json] output format = text
```

### CODE REVIEW

File: src/main/java/bio/terra/cli/command/config/get/Format.java
Comment (jmczerk) [inline src/main/java/bio/terra/cli/command/config/get/Format.java:21]: Not sure whether this matters, but it seems like all of the other enum-driven `terra set config ...` commands seem to use all caps for enum-values.  Not sure it's worth adding an additional enum that maps to the existing (lower case) `FormatOptions` enum, but something to consider for consistency...

```
logging level: OFF, ERROR, WARN, INFO, DEBUG, TRACE, ALL
Browser launch mode: MANUAL, AUTO.
App launch mode: DOCKER_CONTAINER, LOCAL_PROCESS.
```

Comment (jaycarlton) [reply src/main/java/bio/terra/cli/command/config/get/Format.java:21]: The easiest fix would be to change the enum to be uppercase. I don't think any legacy artifacts depend on the precise spelling.

Comment (jaycarlton) [reply src/main/java/bio/terra/cli/command/config/get/Format.java:21]: Actually, that breaks people who expect to do `--format=json` out of habit. Instead, @marikomedlock recommends adding the fixup of that enum to PF-1075.

File: src/test/java/integration/Config.java
Comment (marikomedlock) [inline src/test/java/integration/Config.java:104]: I'm concerned about this assertion returning false positives in the future. Can you change it to either check that the whole thing's a valid JSON object or maybe check a larger string that includes the property name : value?

File: src/test/java/integration/Config.java
Comment (marikomedlock) [inline src/test/java/integration/Config.java:92]: Can you please also add a new method to the `unit.Config` test to check that your property is getting set and returned properly (like the `appLaunch()` method but for `format`)?

File: src/main/java/bio/terra/cli/serialization/userfacing/UFFormatConfig.java
Comment (marikomedlock) [inline src/main/java/bio/terra/cli/serialization/userfacing/UFFormatConfig.java:9]: I don't think you need a separate `UF` class for a single property. 2 reasons:
- The other config properties just print their value (e.g. see the get command for `app-launch` and `browser` properties). This would make `format` inconsistent.
- Having a single "format" key set to the config value doesn't provide any information the caller doesn't already have. They know it's the format because they ran the command `terra config get format`.

**Review by marikomedlock — CHANGES_REQUESTED** [review]
Please also modify the `terra config list` command and associated `UFConfig` class to include your new property.

File: src/main/java/bio/terra/cli/command/config/get/Format.java
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/command/config/get/Format.java:10]: 👍🏻

File: src/main/java/bio/terra/cli/businessobject/Config.java
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/businessobject/Config.java:49]: Set in the nullary constructor instead.

File: src/test/java/integration/Config.java
Comment (jaycarlton) [reply src/test/java/integration/Config.java:104]: done

File: src/test/java/integration/Config.java
Comment (jaycarlton) [reply src/test/java/integration/Config.java:92]: done

File: src/main/java/bio/terra/cli/serialization/userfacing/UFFormatConfig.java
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/serialization/userfacing/UFFormatConfig.java]: 👍🏻

File: src/main/java/bio/terra/cli/command/config/get/Format.java
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/command/config/get/Format.java]: waiting for PF-1075 to merge so I can upcase the enum again.

**Review by marikomedlock — COMMENTED** [review]
Can you also update the README to include the new config commands/property?

File: src/main/java/bio/terra/cli/command/config/get/Format.java
Comment (jaycarlton) [reply src/main/java/bio/terra/cli/command/config/get/Format.java:21]: OK, so just delete `printText`?

**Review by jmczerk — APPROVED** [review]
The code looks OK to me, approving on the assumption that the unit test failure is understood and we get a clean run.

**Review by marikomedlock — APPROVED** [review]
Code changes LGTM. Please make sure you have a clean test run before merging.

**Pull request merged**
